packages/skill-api/src/server/default-payment-options.ts

```diff
@@ -0,0 +1,53 @@
+import Stripe from 'stripe'
+
+type StripeConfig = {
+  stripeSecretKey: string
+  apiVersion: '2020-08-27'
+}
+
+type StripeProvider = {name: 'stripe'; paymentClient: Stripe}
+type StripeProviderFunction = (options: StripeConfig) => StripeProvider
+
+type Paypal = 'paypal-client'
+type PaypalProvider = {name: 'paypal'; paymentClient: Paypal}
+type PaypalProviderFunction = (options: {
+  paypalSecretKey: string
+}) => PaypalProvider
+
+type PaymentProviderOptions = StripeProvider | PaypalProvider
+
+export type PaymentOptions = {
+  providers: {
+    stripe?: StripeProvider
+    paypal?: PaypalProvider
+  }
+}
+
+// define providers here for now,
+// but eventually they can go in a `providers/` directory
+export const StripeProvider: StripeProviderFunction = (config) => {
+  const stripeClient = new Stripe(config.stripeSecretKey, {
+    apiVersion: config.apiVersion,
+  })
+
+  return {
+    name: 'stripe',
+    paymentClient: stripeClient,
```

Comment:
thinking: instead of `paymentClient` pointing to stripe, it gives us an interface that is wrapping a specific client, but would remain the same regardless of the actual client being used under the hood

https://github.com/skillrecordings/products/blob/902c94bb1b7a21879fb2fd206c06146d8fa98935/packages/skill-api/src/core/services/transfer-purchase.ts#L113C1-L118C8

```
const updateStripeCostumer = await stripe.customers.update(
  updateMerchantCustomer.identifier,
  {
    email: targetUserEmail,
  },
)
```

becomes 

```
const updatedCustomer = await paymentClient.updateCustomer(
  updateMerchantCustomer.identifier,
  {
    email: targetUserEmail,
  },
)
```

this is goofy way to capture it, but here's all the current touch points i could find that hit `stripe.*`

![2024-02-20 at 12 15 21@2x](https://github.com/skillrecordings/products/assets/86834/78144157-f083-47cd-b236-ab153b967910)

![2024-02-20 at 12 15 51@2x](https://github.com/skillrecordings/products/assets/86834/03aa850a-8a0d-4080-a871-b2de848bf917)

![2024-02-20 at 12 16 30@2x](https://github.com/skillrecordings/products/assets/86834/fdd49dae-d989-44b4-b5f7-21aff4acdbb8)

![2024-02-20 at 12 16 49@2x](https://github.com/skillrecordings/products/assets/86834/0349e252-0c76-473f-8bc0-169781ecb64d)

![2024-02-20 at 12 17 01@2x](https://github.com/skillrecordings/products/assets/86834/afdd9159-dfa0-45cd-b8ba-c62a29c2af23)